scripts/dangers/keep-package-and-lock-in-sync.js

```diff
@@ -14,6 +14,117 @@
  * permissions and limitations under the License.
  */
 
+import {
+  ValidateHost,
+  ValidateHttps,
+  ValidateScheme,
+  ParseLockfile,
+} from 'lockfile-lint-api';
+
+const defaultLockfilePath = `${process.cwd()}/package-lock.json`;
+const defaultSchemes = ['npm'];
+
+export function lockFileLint(
+  pathToLockFile = defaultLockfilePath,
+  {
+    schemes = defaultSchemes,
+    scheme = false,
+    host = true,
+    protocol = true,
+    basePath = pathToLockFile,
+  } = {}
+) {
+  const options = {
+    lockfilePath: pathToLockFile,
+  };
+
+  const parser = new ParseLockfile(options);
+  const lockfile = parser.parseSync();
+  const validators = [];
+
+  if (host) {
+    validators.push(
+      [new ValidateHost({
+        packages: lockfile.object,
+      }), {
+        onSuccess() {
+          message(`[lock-files] ${basePath} has passed host validation`);
+        },
+      }]
+    );
+  }
+
+  if (protocol) {
+    validators.push(
+      [new ValidateHttps({
+        packages: lockfile.object,
+      }), {
+        onSuccess() {
+          message(`[lock-files] ${basePath} has passed protocol validation`);
+        },
+      }]
+    );
+  }
+
+  if (scheme) {
+    validators.push(
+      [new ValidateScheme({
+        packages: lockfile.object,
+      }), {
+        onSuccess() {
+          message(`[lock-files] ${basePath} has passed scheme validation`);
+        },
+      }]
+    );
+  }
+
+  validators.forEach(([validator, {
+    validate = (opts) => validator.validate(opts),
+    onWarn = (...args) => warn(...args),
+    onError = (...args) => fail(...args),
+    onSuccess = (...args) => message(...args),
+  }]) => {
+    let result;
+    try {
+      result = validate(schemes);
+    } catch (error) {
+      onWarn(error);
+    }
+
+    if (result && result.type === 'success') {
+      onSuccess();
+    } else if (result.type === 'error') {
```

Comment:
also need to check for result here
```suggestion
    } else if (result  && result.type === 'error') {
```